pkgs/top-level/all-packages.nix

```diff
@@ -1823,6 +1824,14 @@ let
     mpi = pkgs.openmpi;
   };
 
+  hdf5-cpp = hdf5.override {
```

Comment:
You should use `appendToName` to give the other `hfd5` packages a distinct name. Otherwise `nix-env` can't tell them apart.